### PR TITLE
C++11 cleanup, p6: Remove ShiftedMatrix(|Generalized)

### PR DESCRIPTION
**Pull request by tamiko**
In reference to #4212

### CODE REVIEW

File: include/deal.II/lac/eigen.h [hunk 1]
```diff
@@ -334,7 +334,9 @@ EigenInverse<VectorType>::solve (double           &value,
   SolverControl::State conv=SolverControl::iterate;
 
   // Prepare matrix for solver
-  ShiftedMatrix <MatrixType> A_s(A, -value);
+  auto A_op = linear_operator(A);
+  double current_shift = -value;
+  auto A_s = A_op + current_shift * identity_operator(A_op);
```
Comment (drwells) [inline include/deal.II/lac/eigen.h]: Nice!

File: include/deal.II/lac/eigen.h [hunk 2]
```diff
@@ -385,13 +387,17 @@ EigenInverse<VectorType>::solve (double           &value,
       // Compute unshifted eigenvalue
       value = (entry * x (i) < 0.) ? -length : length;
       value = 1./value;
-      value -= A_s.shift ();
+      value -= current_shift;
```
Comment (bangerth) [inline include/deal.II/lac/eigen.h]: This must be the most complicated way to write this expression. Would you mind taking the opportunity to rewrite this as
```
  value = (entry * x (i) < 0.) ? -1 : +1)/length - current_shift();
```
or similar?

**Comment by drwells** [comment]
the Contributors page says that, since March 29, @tamiko has added 40 lines and removed 3061 lines. Keep up the good work!

**Comment by bangerth** [comment]
I can beat that: 452 ++ / 3,730 -- :-) Of course, that raises the question what everyone else has been doing over the last week ;-)))

https://github.com/dealii/dealii/graphs/contributors?from=2017-03-29&to=2017-04-07&type=c

**Comment by tamiko** [comment]
@bangerth But my "recontamination rate" (additions / deletions) is much lower :-P

File: include/deal.II/lac/eigen.h (hunk 2 above)
Comment (tamiko) [inline include/deal.II/lac/eigen.h]: @bangerth changed to your suggestion and verified that it still works.

**Pull request merged**